### PR TITLE
Path to VSTest.console for running unit tests

### PR DESCRIPTION
**Pull request by Rickedb**
Since there is no way to use `msbuild` to run unit tests, it's necessary to add path to `VSTest.Console.exe` or `MSTest.exe`.

But made it only importing **VSTest** because **MSTest** is legacy as described [here](https://docs.microsoft.com/en-us/previous-versions/ms182486(v=vs.140)) and VSTest has better performance.

### CODE REVIEW

**Comment by MarkIngramUK** [comment]
Looking forward to seeing `vstest` integrated with this action 👍

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
       id: setup_msbuild_fallback
       uses: ./
       env:
-        PATH: ''
+        PATH: ''
```
Comment (heaths) [inline .github/workflows/test.yml:34]: Nit: extraneous whitespace.

File: src/main.ts [hunk 2]
```diff
@@ -68,7 +69,7 @@ async function run(): Promise<void> {
         let toolPath = path.join(
           installationPath,
           'MSBuild\\Current\\Bin\\MSBuild.exe'
-        )
+        )
```
Comment (heaths) [inline src/main.ts:72]: Nit: extraneous whitespace.

File: src/main.ts [hunk 3]
```diff
@@ -83,6 +84,11 @@ async function run(): Promise<void> {
           }
         }
 
+        foundTestToolPath = path.join(
```
Comment (heaths) [inline src/main.ts:87]: Probably should verify this actually exists. Just because VS is installed doesn't mean the test runner is.

**Review by heaths — COMMENTED** [review]
I don't think this should be combined with the action for setting up MSBuild. VSTest is a separate component and may not even be installed. It also sets a precedent for just making this one action set up everything, which will vary from machine to machine. On hosted agents maybe this isn't a problem, but on custom agents it could be. VS also installs a huge amount of tools and everyone will want to start adding their paths to this action as well. If that's desirable, perhaps a new action - e.g. setup-vs - might be more apt for those who want to keep their build systems and PATH clean.

**Comment by Rickedb** [comment]
@heaths It does make sense, I agree with you that the action must be as light as possible.

However, in the actual code, it ignores if the `vstest` does not exists, just warning that it wasn't possible to find at the end:
``` typescript
if (!foundTestToolPath) {
  core.warning('Unable to find VSTest.console')
}
```
 If it will gerenerate a new action, should we set it as failed when vstest is missing?

**Comment by heaths** [comment]
`foundTestToolPath` is merely set to the output of `path.join`, which doesn't check for existence. I see that an exists check is done for msbuild, but not vstest.console. In fact, on that note, now that I expanded the unchanged areas of the source, it's a weird place to insert the tool - right be `foundToolPath` (for msbuild) is assigned from the local temp variable `toolPath`.

**Comment by timheuer** [comment]
This actually reminded me that https://github.com/marketplace/actions/setup-vstest-console-exe exists.  Going to close this out to keep it focused on MSBuild

**Pull request closed without merging**